cdk/lib/support-workers.ts

```diff
@@ -109,18 +100,67 @@ export class SupportWorkers extends GuStack {
       },
     };
 
-    const createLambda = (
+    const createScalaLambda = (
       lambdaName: string,
       additionalPolicies: PolicyStatement[] = []
     ) => {
       const lambdaId = `${lambdaName}Lambda`;
       const lambda = new GuLambdaFunction(this, lambdaId, {
         ...lambdaDefaultConfig,
+        fileName: `support-workers.jar`,
+        runtime: Runtime.JAVA_21,
         handler: `com.gu.support.workers.lambdas.${lambdaName}::handleRequest`,
         functionName: `${this.stack}-${lambdaName}Lambda-${this.stage}`,
-        environment: {
-          ...lambdaDefaultConfig.environment,
-        },
+        initialPolicy: [
+          s3Policy,
+          cloudWatchLoggingPolicy,
+          ...additionalPolicies,
+        ],
+      });
+      this.overrideLogicalId(lambda, {
+        logicalId: lambdaId,
+        reason: "Moving existing lambda to CDK",
```

Comment:
this probably doesn't make sense any more does it?

Reply:
The comment? I think it's roughly accurate, what else would we put?

Reply:
so it's a permanent change? (I realise it was not added from your PR)
In my mind I thought it needed to be overridden during the switch over to avoid clashes and then it could be removed?
Maybe if it's needed to match up with log group names or state machine definitions it would be something like
`"Align function name with state machine definition"`?